ui/components/multichain/network-list-menu/network-list-menu.tsx

```diff
@@ -297,7 +297,7 @@ export const NetworkListMenu = ({ onClose }: { onClose: () => void }) => {
           }
 
           if (permittedAccountAddresses.length > 0) {
-            addPermittedChain(selectedTabOrigin, network.chainId);
+            dispatch(addPermittedChain(selectedTabOrigin, network.chainId));
```

Comment:
Fixing this upstream in develop here https://github.com/MetaMask/metamask-extension/pull/28741

Reply:
nevermind, this is the result of a bad merge against develop. Develop actually properly dispatches this

Reply:
resolved in the CAIP-25 permission branch upstream of this